app/views/UserGroup/AddUsergroupModal/index.tsx

```diff
@@ -94,55 +125,131 @@ function AddUserGroupModal(props: Props) {
 
     const error = getErrorObject(riskyError);
 
-    const {
-        pending: pendingAddUserGroup,
-        trigger: triggerAddUserGroup,
-    } = useLazyRequest<UserGroup, UserGroupAdd>({
-        url: isDefined(initialValue?.id)
-            ? `server://user-groups/${initialValue?.id}/`
-            : 'server://user-groups/',
-        method: isDefined(initialValue?.id)
-            ? 'PATCH'
-            : 'POST',
-        body: (ctx) => ctx,
-        onSuccess: (response) => {
-            onSuccess(response.id);
-            onModalClose();
-            if (isDefined(initialValue?.id)) {
-                alert.show(
-                    'Successfully edited user group.',
-                    { variant: 'success' },
-                );
-            } else {
-                alert.show(
-                    'Successfully created user group.',
-                    { variant: 'success' },
-                );
-            }
+    const [
+        createUsergroup,
+        {
+            loading: createUsergroupPending,
         },
-        onFailure: () => {
-            if (isDefined(initialValue?.id)) {
+    ] = useMutation<UserGroupCreateMutation, UserGroupCreateMutationVariables>(
+        USER_GROUP_CREATE,
+        {
+            onCompleted: (response) => {
+                if (!response?.userGroupCreate?.ok) {
+                    return;
+                }
+
+                const responseId = response?.userGroupCreate?.result?.id;
+                if (responseId) {
+                    onSuccess(responseId);
+                }
+                const {
+                    ok,
+                    errors,
+                } = response && response.userGroupCreate;
+
+                if (errors) {
+                    const formError = transformToFormError(removeNull(errors) as ObjectError[]);
+                    setError(formError);
+                    alert.show(
+                        'Failed to create user group.',
+                        { variant: 'error' },
+                    );
+                } else if (ok) {
+                    alert.show(
+                        'Successfully created user group!',
+                        { variant: 'success' },
+                    );
+                }
+                onModalClose();
```

Comment:
We should try to not call too many callbacks.

Also, we are not handling the case when error occurs.